pkg/store/cache/cache.go

```diff
@@ -72,6 +73,13 @@ func newCommonMetrics(reg prometheus.Registerer) *commonMetrics {
 			Name: "thanos_store_index_cache_hits_total",
 			Help: "Total number of items requests to the cache that were a hit.",
 		}, []string{"item_type"}),
+		dataSizeBytes: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
Can we expose native histograms too?

Reply:
Tbh I never used that before. Do I just need to configure `NativeHistogramBucketFactor` to enable it? Can I still keep the pre defined buckets if I am not using native histogram?

Reply:
Yes, that is the way to do it. The server will then decide which version to scrape based on how it is configured.

However on second thought, let's maybe enable native histograms in a single pass in a separate PR. It would be strange to have some metrics in conventional and other in native histograms.